zerver/tests/test_messages.py

```diff
@@ -2734,10 +2734,16 @@ def test_messages_for_ids(self) -> None:
 
 class MessageVisibilityTest(ZulipTestCase):
     def test_get_first_visible_message_id_with_no_limit(self) -> None:
+        Message.objects.all().delete()
+        for i in range(5):
+            self.send_stream_message(self.example_email("othello"), "Scotland")
+
         realm = get_realm("zulip")
         realm.message_visibility_limit = None
         realm.save()
 
+        with mock.patch('zerver.lib.message.estimate_recent_messages', return_value=5):
+            maybe_update_first_visible_message_id(realm, 30)
```

Comment:
So, one great thing about separating the "maybe" logic out from the actual computation and update: you can skip this complexity in tests! Just call `update_first_visible_message_id` directly, rather than try to fool the "maybe" (which e.g. might have to change as we change the "maybe" logic.)